spy.go

```diff
@@ -42,3 +103,65 @@ func Execute(w io.Writer, commandLine string) error {
 
 	return nil
 }
+
+// NewSession creates and returns a new shell session with a file
+func NewSession(filepath string, opts ...Option) (session, error) {
+	s := session{}
+	f, err := os.Create(filepath)
+	if err != nil {
+		return session{}, err
+	}
+
+	s.Input = os.Stdin
+	s.Terminal = os.Stdout
+	s.Recorder = f
+
+	for _, opt := range opts {
+		err := opt(&s)
+		if err != nil {
+			return session{}, err
+		}
+	}
+	return s, nil
+}
+
+func (s *session) Run() error {
+	scanner := bufio.NewScanner(s.Input)
+
+	user, err := user.Current()
+	if err != nil {
+		return fmt.Errorf("retrieving current user returned err: %v", err)
+	}
+
+	host, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("retrieving host name returned err: %v", err)
+	}
+
+	prompt := fmt.Sprintf("%v@%v:~$", user.Username, host)
+
+	if s.ShellPrompt != "" {
+		prompt = s.ShellPrompt + ":~$"
+	}
+
+	if s.TimestampMode {
+		s.RecordTime(time.Now())
```

Comment:
This feels like it would be better as something like:

```go
s.Timestamp(time.Now())
```

and that method would simply no-op if timestamps are disabled.